File: Arrays/Two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n=len(nums)
        b={}
        for i in range(n):
            diff=target-nums[i]
            if(diff in b):
                return [i,b[diff]]
            else:
                b[nums[i]]=i
        return []
        

File: Arrays/Two_sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
       int n=nums.length,i;
       HashMap<Integer,Integer> map=new HashMap<Integer,Integer>();
       for(i=0;i<n;i++){
           int diff=target-nums[i];
           if(map.containsKey(diff)==true){
               return new int[]{i,map.get(diff)};
           }
           else{
               map.put(nums[i],i);
           }
       }
       return new int[]{0,0};
    }
}